dot-net/Centroid/Config.cs

```diff
@@ -78,7 +78,23 @@ public override bool TryConvert(ConvertBinder binder, out object result)
 
             try
             {
-                result = Convert.ChangeType(RawConfig, binder.Type);
+                JToken rawConfig = RawConfig;
+                result = rawConfig.ToObject(binder.Type);
+                //ensure at least one property in the config bound to the type
```

Comment:
I wonder if, for the purposes of checking the failed binding, you could just compare it with default(T) (or more specifically, with whatever the equivalent code would be given that you have a Type)